src/help.rs

```diff
@@ -132,26 +135,69 @@ impl Help {
         self.state.select(Some(i));
     }
 
-    pub fn render(&mut self, frame: &mut Frame, color_mode: ColorMode) {
+    pub fn render(&mut self, frame: &mut Frame) {
         let block = help_rect(frame.size());
 
         self.block_height = block.height as usize;
-        let widths = [Constraint::Length(20), Constraint::Fill(1)];
+
+        let style = match self.config.color_mode {
+            ColorMode::Light => Style::default().fg(Color::Black),
+            _ => Style::default().fg(Color::White),
+        };
+
+        let narrow_mode = frame.size().width < self.config.small_layout_cols;
+        let row_title_width = 20;
+        let table_block_padding = if narrow_mode { 0 } else { 2 };
+
+        let widths = [Constraint::Length(row_title_width), Constraint::Fill(1)];
+        let max_row_detail_length = (
+            block.width - row_title_width
+            - 3 * table_block_padding // Cell padding
+            - 3
+            // Borders
+        ) as usize;
+
         let rows: Vec<Row> = self
             .keys
             .iter()
-            .map(|key| {
-                Row::new(vec![key.0.to_owned(), key.1.into()]).style(match color_mode {
-                    ColorMode::Dark => Style::default().fg(Color::White),
-                    ColorMode::Light => Style::default().fg(Color::Black),
-                })
+            .flat_map(|key| {
+                // Split row details into new rows to emulate line wrap
+                let mut lines = Vec::new();
+                let mut remainder = key.1;
+
+                // Keep splitting
+                while remainder.len() > max_row_detail_length {
+                    let (line, rest) = remainder.split_at(max_row_detail_length);
+                    lines.push(line.to_owned());
+                    remainder = rest;
+                }
+
+                // Add the last split
+                lines.push(remainder.to_owned());
+
+                // Create a row for each line
+                let mut rows = Vec::new();
+
+                // First row: key.0 and first line of key.1
+                if !lines.is_empty() {
+                    rows.push(
+                        Row::new(vec![key.0.to_owned(), Cell::from(lines[0].clone())]).style(style),
+                    );
+                }
+
+                // Rest of rows: only remaining lines of key.1
+                for line in lines.iter().skip(1) {
+                    rows.push(Row::new(vec!["".to_owned(), line.clone()]).style(style));
+                }
+
+                rows
```

Comment:
I found this code not easy to understand. is there a way to make it more simple ?

Reply:
Added some more comments.